app-modules/ai/tests/Tenant/Unit/Jobs/CloneAiThreadTest.php

```diff
@@ -0,0 +1,87 @@
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use AdvisingApp\Ai\Enums\AiApplication;
+use AdvisingApp\Ai\Enums\AiModel;
+use AdvisingApp\Ai\Jobs\CloneAiThread;
+use AdvisingApp\Ai\Models\AiAssistant;
+use AdvisingApp\Ai\Models\AiMessage;
+use AdvisingApp\Ai\Models\AiThread;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use App\Models\User;
+use Filament\Notifications\DatabaseNotification;
+use Illuminate\Support\Facades\Notification;
+
+it('can clone a thread and its messages', function () {
+    Notification::fake();
+
+    $sender = User::factory()->licensed(LicenseType::cases())->create();
+    $recipient = User::factory()->licensed(LicenseType::cases())->create();
+
+    expect($recipient->aiThreads)->toHaveCount(0);
+
+    $assistant = AiAssistant::factory()->create([
+        'application' => AiApplication::Test,
+        'is_default' => true,
+        'model' => AiModel::Test,
+    ]);
+
+    $thread = AiThread::factory()
+        ->for($assistant, 'assistant')
+        ->has(AiMessage::factory()->count(3), 'messages')
+        ->for($sender, 'user')
+        ->create();
+
+    $originalMessages = $thread->messages;
+    dispatch(new CloneAiThread($thread, $sender, $recipient));
+
+    $recipient->refresh();
+    $clonedThread = $recipient->aiThreads()->latest()->first();
+
+    expect($clonedThread)->not->toBeNull();
+    expect($clonedThread->id)->not->toBe($thread->id);
+
+    expect($clonedThread->name)->toBe($thread->name);
+    expect($clonedThread->assistant_id)->toBe($thread->assistant_id);
+    expect($clonedThread->folder_id)->toBe($thread->folder_id);
```

Comment:
Ah, @danharrin if this is true. Do we have a problem?

@kandarp-canyon this is not something you need to fix right now. But if this is true then there is something about cloning threads we need to change.
Folders should be owned by Users, and if cloned threads are keeping folder ids then that isn't good.